### ISSUE TITLE
False positive when method-scoped covariant type variable is used in parameter annotation

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#8614]
If a class-scoped type variable is explicitly marked as covariant, it cannot be safely used in a parameter annotation, and pyright correctly flags this. However, if the same type variable is scoped to the method rather than the class, then variance is meaningless, and no such error should be emitted.

```python
from typing import Generic, TypeVar

S = TypeVar("S", covariant=True)
T = TypeVar("T", covariant=True)


class C(Generic[T]):
    # Error is correct here
    def meth1(self, x: T) -> T:
        ...

    # This should not result in an error
    def meth2(self, x: S) -> S:
        ...

```

**Comment by erictraut** [comment]
This is included in pyright 1.1.375.
